### PR TITLE
chore: documentation for the jinja command.

### PR DESCRIPTION
**Pull request by butler54**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: docs/trestle_author_jinja.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Trestle author jinja - output templating support for oscal documents.
+
+Unfortunately OSCAL documents are not yet universally accepted. Therefore to support various OSCAL and non-OSCAL compliance workflows `trestle author jinja` is designed to provide end users with the ability to use jinja to produce customized output. This complements the more structured commands `trestle author catalog-{assemble|generate}`, `trestle author profile-{assemble|generate}` and `trestle author ssp-{assemble|generate}` and allows arbitrary use of jinja.
+
+## Jinja and jinja extensions provided by trestle.
+
+[Jinja](https://jinja.palletsprojects.com/en/3.0.x/) is a powerful templating engine that is both more flexible that pure 'Moustache' approaches, and not coupled to a particular web application server. (as an example Django templates). Users are encouraged to review the [template designer documentation](https://jinja.palletsprojects.com/en/3.0.x/templates/) for jinja as all core functionality is exposed.
```
Comment (enikonovad) [inline docs/trestle_author_jinja.md:7]: extra dot after `web application server`

File: docs/trestle_author_jinja.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Users are recommended to use homebrew to install the latest python 3 and then install python within a venv.
+```
+
+Users are free to use the LUT to inject more complex variables (arrays of data etc) to use at their own will using standard jinja templating.
+
+### Resolved catalog interface (profile)
+
+Passing `-p` exposes a catalog, resolved from the profile, at `catalog` and a `trestle.core.catalog_interface.CatalogInterface` at `catalog_interface`.
+
+This allows user to perform various task such as iterating ove reach group and printing the group title.
```
Comment (enikonovad) [inline docs/trestle_author_jinja.md:74]: over

File: docs/trestle_author_jinja.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+## CLI invocation
+
+Note the examples here use markdown, however, jinja can quite easily target xml or html if used w/o specific markdown content.
+`trestle author jinja -i input_template.md.jinja -o output_file.md -ssp SSP_NAME -p PROFILE_NAME -lut lookup_table.yaml -elp lut.prefix`
+
+- `-i` input file path, relative to cwd. Users are encouraged to use the `file_name.target_extension.jinja` best practice as it helps mitigate issues, however is not required.
+- `-o` final output path, relative to cwd.
+- `-ssp` (optional) ssp name (in the trestle project). When used a interface will be made available to the jinja template.
+- `-p` (optional) profile name (in the trestle project). When used a interface to the resolved catalog will be made available to the jinja template.
```
Comment (enikonovad) [inline docs/trestle_author_jinja.md:27]: I think from the end-user perspective it is a bit unclear what will be printed if those flags are passed. While the sections below provide some insights into what will be done if those flags are passed, but not sure if it is sufficient. Maybe adding a section like `if you pass -ssp then the following structure will be followed for each control...`

**Review by enikonovad — APPROVED** [review]
LGTM, just a small comment on the `-ssp` and `-p` flags

File: docs/trestle_author_jinja.md (hunk 1 above)
Comment (butler54) [reply docs/trestle_author_jinja.md:7]: Done

File: docs/trestle_author_jinja.md (hunk 3 above)
Comment (butler54) [reply docs/trestle_author_jinja.md:27]: I've changed this. I've also added a section for sample jinja templates.

Comment (enikonovad) [reply docs/trestle_author_jinja.md:27]: Great thanks will give it a read once pushed

**Pull request merged**